### PR TITLE
Issue #3954: MethodParamPadCheck input files split

### PR DESCRIPTION
**Pull request by GitToasterhub**
Issue #3954 Split and orginise inputs by MethodParamPad Test.

### CODE REVIEW

**Comment by GitToasterhub** [comment]
@romani @rnveach 
Commit c02034b0cd968325910cbcce3149f84f2daabd72 message: "Issue #3954:MethodParamPad Test input split\n" is invalid
Proper commit message should adhere to the following rules:
    1) Must match one of the following patterns:
        ^Issue #\d+: .*$
        ^Pull #\d+: .*$
        ^(minor|config|infra|doc|spelling): .*$
    2) It contains only one line of text
    3) Must not end with a period, space, or tab
    4) Commit message should be less than or equal to 200 characters
It happens again,but I changed commit message to Issue #3954:MethodParamPad Test input split.
That seems to be valid,but is not

**Comment by romani** [comment]
read attentively format, space after ":" is required.

**Comment by GitToasterhub** [comment]
@romani Thanks so much for help!I spent so much time on it just because i hadn't noticed that space..much more than on issue

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package com.puppycrawl.tools.checkstyle.checks.whitespace.methodparampad;
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java]: this input is used only in this Check, so file should be moved (not copied).

Comment (GitToasterhub) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java]: You mean that,I should placce in created folder shortcuts only,not the class?

Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java]: you need to do "git mv oldpathname newpathname", no a simple copy of file.

Comment (GitToasterhub) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java]: Is it OK now?

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java [hunk 2]
```diff
@@ -80,4 +80,4 @@ public void newArray()
         {
         }
     }
-}
+}
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java]: please restore newline at the end of file

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java (hunk 1 above)
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java]: @GitToasterhub Yes.
If you look at https://github.com/checkstyle/checkstyle/pull/3983/files , you ca see the file head has the `->` arrow. This signifies the file moved to a new location.

**Comment by GitToasterhub** [comment]
@rnveach  Oh,it seems that InputWhitespaceAround follows the pattern just like InputMethodParamPad,isn't it? Also I copied InputWhitespaceAround as it is used by other tests as well.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java (hunk 2 above)
Comment (GitToasterhub) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java]: Added new line

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputWhitespaceAround.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package com.puppycrawl.tools.checkstyle.checks.whitespace.methodparampad;
+
+@SuppressWarnings({"this", "that"})
+public class InputWhitespaceAround
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputWhitespaceAround.java]: Class and file name are still incorrect.
Input should be named after test. Test is named `MethodParamPadCheckTest`, Input should be named `InputMethodParamPad...`.

Comment (GitToasterhub) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputWhitespaceAround.java]: Do you mean InputWhitespaceAround should be named InputMethodParamPadWhitespaceAround?

We split checks into separate files,so even if 2 checks have the same name input file they wouldn't depend on 1 file because they are in different namespaces

Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputWhitespaceAround.java]: > Do you mean InputWhitespaceAround should be named InputMethodParamPadWhitespaceAround?

yes.

>  they are in different namespaces

This is another problem we have. Multiple input files are the exact same so users get confused and make changes to the wrong file. This is why we want file names to be named after the test and in it's own directory.

**Comment by GitToasterhub** [comment]
@rnveach I finally have got what you meant and changed.Hope now it's ok.

**Comment by GitToasterhub** [comment]
oh.. wrecker /build is failed.. I can't look the details of it

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java [hunk 4]
```diff
@@ -79,5 +79,5 @@ public void newArray()
             ()
         {
         }
-    }
+    }
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java]: please revert this change

**Comment by romani** [comment]
wercker log is the same as https://gist.github.com/romani/306d36ba1a6590658db911906b83ac66

please run wercker commands on local https://github.com/checkstyle/checkstyle/blob/master/wercker.yml#L172 and share output.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java (hunk 4 above)
Comment (GitToasterhub) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java]: Oh.It's a truly new level of development for me.Everything metters even spaces that visually and practically doesn't make a difference actually is important.I should get used to...I spent for task that could be done in 5-10 minutes few days.. xD

Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPad.java]: you need to practicing to deliver only requested changes. You need to review your changes/PR yourself to verify that all is fine.
Messing all together is bad practice in most teams.
